flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -711,7 +713,7 @@ public long getCheckpointIdOfIgnoredInFlightData() {
          * <p>Note that checkpoint state is always kept if the job terminates with state {@link
          * JobStatus#FAILED}.
          */
-        DELETE_ON_CANCELLATION(true),
+        DELETE_ON_CANCELLATION,
```

Comment:
Since we're at it, we could have the enum implement `DescribedEnum` and clean up the description of the config option to not mention the enum values in the text (this will then be generated from the enum).